server/handler.go

```diff
@@ -2024,42 +2024,46 @@ func (h *tablesInsertHandler) Handle(ctx context.Context, r *tablesInsertRequest
 	)
 	encodedTableData, err := json.Marshal(table)
 	if err != nil {
-		return nil, err
+		return nil, errInternalError(err.Error())
 	}
 	var tableMetadata map[string]interface{}
 	if err := json.Unmarshal(encodedTableData, &tableMetadata); err != nil {
-		return nil, err
+		return nil, errInternalError(err.Error())
 	}
 
 	conn, err := r.server.connMgr.Connection(ctx, r.project.ID, r.dataset.ID)
 	if err != nil {
-		return nil, err
+		return nil, errInternalError(err.Error())
 	}
 	tx, err := conn.Begin(ctx)
 	if err != nil {
-		return nil, err
+		return nil, errInternalError(err.Error())
 	}
 	defer tx.RollbackIfNotCommitted()
 	if r.table.Schema != nil {
 		if err := r.server.contentRepo.CreateTable(ctx, tx, r.table); err != nil {
-			return nil, err
+			return nil, errInternalError(err.Error())
 		}
 	}
+	tableID := r.table.TableReference.TableId
+	if exists := r.dataset.HaveTable(tableID); exists {
+		return nil, errDuplicate(fmt.Errorf("table %s is already created", tableID).Error())
+	}
 	if err := r.dataset.AddTable(
 		ctx,
 		tx.Tx(),
 		metadata.NewTable(
 			r.server.metaRepo,
 			r.project.ID,
 			r.dataset.ID,
-			r.table.TableReference.TableId,
+			tableID,
 			tableMetadata,
 		),
 	); err != nil {
-		return nil, err
+		return nil, errInternalError(err.Error())
```

Comment:
The following logic looks good.

Create a variable like `var ErrDuplicatedTable = errors.New("table is already created")` in the `internal/metadata` package and then return `fmt.Errorf("table %s: %w", table.ID, ErrDuplicatedTable)`, and return `errDuplicate` if error contains `metadata.ErrDuplicatedTable` by `errors.As` in the handler side.

Reply:
If possible, could you please add a Go's test case for this problem to `server_test.go` ?

Reply:
Thanks for helpful comments.
I will modify it within a few days.

Reply:
> by `errors.As` in the handler side

Is this mistake of `errors.Is`?

I modifyied code as such.

Reply:
I made a mistake 😅  . `errors.Is` is correct!